### DISCUSSION TITLE
Weight-based Dose Brackets in PK-Sim

### DISCUSSION THREAD

**Discussion by Bcicali** [Open-Systems-Pharmacology/Forum#1142] [Questions & Problems]
Hi All,

I am working on recreating a clinical study simulation for model validation purposes and the study utilized an weight-based dose bracketing approach; e.g. >60kg = 100mg, 45-60 kg = 75mg, 35-44 kg= 50 mg, etc. 

Is there a way to create this type of dosing schema for a PK-Sim population simulation?

Thank you!
Brian

**Comment by msevestre** [comment]
Hi @Bcicali 
I don't think there is an easy way to do this in PK-Sim....
But you can easily do this in R using ospsuite-r by creating a user-defined variability for the Dose parameter of your application

**Comment by Yuri05** [comment]
As @msevestre said, there is no easy way do to it in PK-Sim (would be a nice feature I think).

One possibility would be to create an individual simulation in PK-Sim with arbitrary dose, export it to MoBi and set Bodyweight-range-based dose in MoBi; then import this simulation in PK-Sim and combine it with your population.
For this:
1. Create an individual simulation in PK-Sim with the dose e.g. 1mg/kg and send this simulation to MoBi (after that, delete this simulation in PK-Sim)
2. In MoBi: open the Events building block, go to the first application and go to the parameters of the **ProtocolSchemaItem** container
![grafik](https://user-images.githubusercontent.com/25061876/155147018-bd082a38-c2a5-4957-a32b-0e6b3fde8bc2.png)
3. Add new dosing parameters for each weight range (e.g. `Dose_60`, `Dose_45`, etc.):
<img width="400" src="https://user-images.githubusercontent.com/25061876/155122220-6e6fe018-094e-4ce7-9f6d-29c5ac735823.png">

Then define the `Dose` formula like below using the newly introduced parameters (make sure you modify the **Dose** parameter and NOT _DosePerBodyWeight_ etc.)

![grafik](https://user-images.githubusercontent.com/25061876/155123083-211a5c47-5057-4768-a077-94528bf98c02.png)

  * The notation `Condition ? X : Y` means: if `Condition `applies: calculate `X`; otherwise calculate `Y` 
     Thus the formula above is a **nested conditional statement**:
```
     IF BodyWeight > 60
       Dose = ...
     ELSE IF BodyWeight >= 45
       Dose = ...
     ELSE IF BodyWeight >=35
     ...
```

5. Activate the "**Advanced parameter**" checkbox for parameters `Dose`, `DosePerBodyWeight` and `DosePerBodySurfaceArea`
![grafik](https://user-images.githubusercontent.com/25061876/155143424-13d750a2-0e0e-49de-bd07-8de2061cffb3.png)
4. Update the simulation from the changed events building block
![grafik](https://user-images.githubusercontent.com/25061876/155123258-0edb5324-655c-4f90-88c4-61614a30ad10.png)
5. Save simulation as pkml
6. Go back to PK-Sim, go to Import/Export and select Import population simulation
![grafik](https://user-images.githubusercontent.com/25061876/155123675-5a6c124f-e911-4f85-b282-35a9a724a251.png)
7. Select the pkml simulation exported from MoBi and combine it with your population of interest
![grafik](https://user-images.githubusercontent.com/25061876/155124178-c76adf02-f1ca-46f6-b520-abb911f8cfe8.png)
Now you can use you weight-range-based dosing in PK-Sim.
![grafik](https://user-images.githubusercontent.com/25061876/155145478-0b1352af-27d2-420a-b94c-aecf3677e38e.png)

Attached are example projects in PK-Sim and MoBi using this approach
[Example.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/8117113/Example.zip)

**Comment by Yuri05** [comment]
Another possibility: use the R package [ospsuite.reportingengine](https://www.open-systems-pharmacology.org/OSPSuite.ReportingEngine/) and use the "Study Design" feature.
For that, you would need to create a "Study Design" file in the CSV format - which allows arbitrary conditional extensions of a population (e.g. weight-range-based dosage)

![grafik](https://user-images.githubusercontent.com/25061876/155129686-4db32050-7fcb-418c-b442-269a865566ea.png)

**For each application** you need to add a target column with the Dose of the corresponding application, e.g. for the example from the previous post it would be just 1 target column `Applications|App1|Application_1|ProtocolSchemaItem|Dose`

Then:
* export your population from PK-Sim as csv
* start R and 
  * load the exported population via [loadPopulation](https://www.open-systems-pharmacology.org/OSPSuite-R/reference/loadPopulation.html)
  * extend your population with created study design via [addStudyParameters](https://www.open-systems-pharmacology.org/OSPSuite.ReportingEngine/reference/addStudyParameters.html)
  * save the modified population to CSV
* import the modified population from CSV in PK-Sim

**Reply by Bcicali** [reply]
Thank you both very much for your quick responses and insight!

I will try implementing both methods this week and will circle back.

**Comment by hofferty** [comment]
Thank you very much for this method @Yuri05 . I face some problems when I try to install the ospsuite.reportengine. 
After installing the required packages and downloading the ospsuite.reportengine, I can't find it in the library (error).
With which version did you run this package? I work with R 4.1.2
Thank you,
Yannick

**Reply by Yuri05** [reply]
@hofferty Please open a new discussion for that. Having 1 topic per thread makes the communication much more easy and transparent.
